lib/selinux/selinux_linux.go

```diff
@@ -0,0 +1,145 @@
+/*
+ * Teleport
+ * Copyright (C) 2025  Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package selinux
+
+import (
+	"bufio"
+	"bytes"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/gravitational/trace"
+	ocselinux "github.com/opencontainers/selinux/go-selinux"
+)
+
+const (
+	selinuxConfig        = "/etc/selinux/config"
+	selinuxTypeTag       = "SELINUXTYPE"
+	permissiveModuleName = "permissive_" + domain
+)
+
+// copied from github.com/opencontainers/selinux/go-selinux/selinux-linux.go
+func readConfig(target string) string {
+	in, err := os.Open(selinuxConfig)
+	if err != nil {
+		return ""
+	}
+	defer in.Close()
+
+	scanner := bufio.NewScanner(in)
+
+	for scanner.Scan() {
+		line := bytes.TrimSpace(scanner.Bytes())
+		if len(line) == 0 {
+			// Skip blank lines
+			continue
+		}
+		if line[0] == ';' || line[0] == '#' {
+			// Skip comments
+			continue
+		}
+		fields := bytes.SplitN(line, []byte{'='}, 2)
+		if len(fields) != 2 {
+			continue
+		}
+		if bytes.Equal(fields[0], []byte(target)) {
+			return string(bytes.Trim(fields[1], `"`))
+		}
+	}
+	return ""
+}
+
+// CheckConfiguration returns an error if SELinux is not configured to
+// enforce the SSH service correctly.
+func CheckConfiguration() error {
+	if !ocselinux.GetEnabled() {
+		return trace.Errorf("SELinux is disabled or not present")
+	}
+	if ocselinux.EnforceMode() != ocselinux.Enforcing {
+		return trace.Errorf("SELinux mode is not enforcing, SELinux will not constrain anything")
+	}
+
+	selinuxType := readConfig(selinuxTypeTag)
+	if selinuxType == "" {
+		return trace.NotFound("could not find SELinux type")
+	}
+	selinuxDir := filepath.Join("/var/lib/selinux", selinuxType, "active/modules")
```

Comment:
Is this always the same path on all supported systems? Can this path be retrieved by querying selinux?

Reply:
Yep, this path will largely be the same however the `selinuxType` is retrieved so it works for the current system